src/controllers/GatewayController.ts

```diff
@@ -150,4 +156,29 @@ export default class GatewayController {
 		if (!userConfig) throw new GatewayError('Not authenticated');
 		this.discoveryQueue.removeFromQueue(userConfig.id);
 	}
+
+	public async onTyping(ws: WebSocket, packet: ClientTypingPacket) {
+		const userConfig = this.authenticatedClients.get(ws);
+		if (!userConfig) throw new GatewayError('Not authenticated');
+
+		const { channelID } = packet.data;
+		const channel = await this.channelService.findOne({ id: channelID });
+		if (!channel) throw new GatewayError('Channel does not exist');
+
+		const gatewayTypingPacket: GatewayTypingPacket = {
+			type: GatewayPacketType.GatewayTyping,
+			data: {
+				userID: userConfig.id,
+				channelID
+			}
+		};
+
+		if (channel instanceof EventChannel) {
+			// Send packet to everyone connected
+			await this.broadcast<GatewayTypingPacket>(gatewayTypingPacket);
+		} else if (channel instanceof DMChannel) {
+			// Send packet to the users in the dm channel
+			await this.sendMessage([...channel.users.map(user => user.id)], gatewayTypingPacket);
```

Comment:
```suggestion
			await this.sendMessage(channel.users.map(user => user.id), gatewayTypingPacket);
```

I don't think copying the array is necessary here

Reply:
Oh right😂🤦‍♂️